spec/type_references_and_ownership.md

```diff
@@ -0,0 +1,295 @@
+       the owner already exists, but is required when it is still being created (possibly in parallel).
+    2. The `name` field of the dependent resource in this mode serves a dual purpose - to name the dependent resource and to identify the parent resources.
+       Each segment of the name corresponds to an owning resource, so creating a Batch Pool `foo` in Batch account `account` would have `name` = `account/foo`.
+
+#### What ASO does
+Dependent resources in ASO have properties which map to the name/path to their owner. For example 
+[MySQLFirewallRuleSpec](https://github.com/Azure/azure-service-operator/blob/92240406aff3863f3a267d8a1dc1e28aa3e841ae/api/v1alpha1/mysqlfirewallrule_types.go#L15) looks like this:
+```go
+type MySQLFirewallRuleSpec struct {
+	ResourceGroup  string `json:"resourceGroup"`
+	Server         string `json:"server"`
+	StartIPAddress string `json:"startIpAddress"`
+	EndIPAddress   string `json:"endIpAddress"`
+}
+```
+
+The `ResourceGroup` and `Server` are references to the owners of this type.
+
+#### What k8s-infra does
+k8s-infra is a bit different in that resource references are in Kubernetes language (namespace + name) and not Azure language (resource group + resource name).
+All resource references are done using a special type `KnownTypeReference`.
+There are two patterns in k8s-infra today.
+
+One pattern is used for ResourceGroup, where 
+[top level resources have a link to the resource group they are in](https://github.com/Azure/k8s-infra/blob/14105b1cb3f6967cd086c9f8f75fb16bb85d6318/apis/microsoft.compute/v1/virtualmachinescaleset_types.go#L323).
+```go
+type VirtualMachineScaleSetSpec struct {
+    // +kubebuilder:validation:Required
+    ResourceGroupRef *azcorev1.KnownTypeReference `json:"resourceGroupRef" group:"microsoft.resources.infra.azure.com" kind:"ResourceGroup"`
+    ...
+}
+```
+
+The other pattern is where the owning resource has links to the dependent resources it is expecting to have:
+```go
+type RouteTableSpecProperties struct {
+    DisableBGPRoutePropagation bool                          `json:"disableBgpRoutePropagation,omitempty"`
+    RouteRefs                  []azcorev1.KnownTypeReference `json:"routeRefs,omitempty" group:"microsoft.network.infra.azure.com" kind:"Route" owned:"true"`
+}
+```
+
+If the dependent resources aren't there, the status of the owning resource reflects an error explaining that.
+
+## Proposal
+
+### How to represent references
+We construct a new type which is our reference holder. I am calling this approach `AugmentedKnownTypeReference` (real type name TBD), because
+it's similar to the current approach that `k8s-infra` is taking but with a few changes.
+
+```go
+type AugmentedKnownTypeReference struct {
+    Kubernetes KubernetesReference
+    Azure      AzureReference
+}
+
+type KubernetesReference struct {
+    // The "Group" of the owning resources GVK. There may be an annotation on the property 
```

Comment:
I've added some details to the document explaining why it isn't needed. The short of it is, the purpose of this is to link to another Kubernetes resource. Because K8s resources are all stored in the storage version in etcd, and all this really is doing is giving us a link to the etcd content (so we can look up the name of the resource in question and then build the ARM ID for it) we don't need the version because it's just an etcd lookup.

Put another way, my understanding is that while there can be different versions of an object in Kubernetes, those versions are just views on the one underlying object - here we only need a link to the one underlying object.